src/Mod/Measure/Gui/ViewProviderMeasureAngle.cpp

```diff
@@ -0,0 +1,340 @@
+            otherSide.Normalize();
+        }
+
+        gp_Vec xAxis = (loc1 - originVector).Normalized();
+        gp_Vec fakeYAxis = (thirdPoint - originVector).Normalized();
+        gp_Vec zAxis = (xAxis.Crossed(fakeYAxis)).Normalized();
+        gp_Vec yAxis = zAxis.Crossed(xAxis).Normalized();
+
+        dimSys = SbMatrix
+        (
+            xAxis.X(), yAxis.X(), zAxis.X(), dimensionOriginPoint.X(),
+            xAxis.Y(), yAxis.Y(), zAxis.Y(), dimensionOriginPoint.Y(),
+            xAxis.Z(), yAxis.Z(), zAxis.Z(), dimensionOriginPoint.Z(),
+            0.0, 0.0, 0.0, 1.0
+        );
+
+        dimSys = dimSys.transpose();
+    }
+
+    return dimSys;
+}
+
+
+
+PROPERTY_SOURCE(MeasureGui::ViewProviderMeasureAngle, MeasureGui::ViewProviderMeasureBase)
+
+
+ViewProviderMeasureAngle::ViewProviderMeasureAngle()
+{
+    sPixmap = "umf-measurement";
+
+    // Primary Arc
+    Gui::ArcEngine *arcEngine = new Gui::ArcEngine();
+    arcEngine->angle.connectFrom(&fieldAngle);
+
+    auto calculatorRadius = new SoCalculator();
+    calculatorRadius->A.connectFrom(&pDragger->translation);
+    calculatorRadius->expression.setValue("oa=length(A)");
+    arcEngine->radius.connectFrom(&calculatorRadius->oa);
+    arcEngine->deviation.setValue(0.1f);
+
+    SoCoordinate3 *coordinates = new SoCoordinate3();
+    coordinates->point.connectFrom(&arcEngine->points);
+
+    SoLineSet *lineSet = new SoLineSet();
+    lineSet->ref();
+    lineSet->vertexProperty.setValue(coordinates);
+    lineSet->numVertices.connectFrom(&arcEngine->pointCount);
+    lineSet->startIndex.setValue(0);
+
+    pLineSeparator->addChild(lineSet);
+
+    // Secondary Arc, from midpoint to label
+    auto engineAngle = new SoCalculator();
+    engineAngle->A.connectFrom(&arcEngine->midpoint);
+    engineAngle->B.connectFrom(&pLabelTranslation->translation);
+    engineAngle->expression.setValue("tA=normalize(A); tB=normalize(B); oa=atan2(tB[1], tB[0])-atan2(tA[1], tA[0])");
+
+    Gui::ArcEngine *arcEngineSecondary = new Gui::ArcEngine();
+    arcEngineSecondary->radius.connectFrom(&calculatorRadius->oa);
+    arcEngineSecondary->deviation.setValue(0.1f);
+    arcEngineSecondary->angle.connectFrom(&engineAngle->oa);
+
+    // Rotate arc
+    auto engineRotMidpoint = new SoComposeRotationFromTo(); // absolute angle to midpoint
+    engineRotMidpoint->from.setValue(SbVec3f(1.0, 0.0, 0.0)); 
+    engineRotMidpoint->to.connectFrom(&arcEngine->midpoint);
+
+    auto matrixEngine = new SoComposeMatrix();
+    matrixEngine->rotation.connectFrom(&engineRotMidpoint->rotation);
+    auto transformEngine = new SoTransformVec3f();
+    transformEngine->matrix.connectFrom(&matrixEngine->matrix);
+    transformEngine->vector.connectFrom(&arcEngineSecondary->points);
+
+    SoCoordinate3 *coordinatesSecondary = new SoCoordinate3();
+    coordinatesSecondary->point.connectFrom(&transformEngine->point);
+    
+    SoLineSet *lineSetSecondary = new SoLineSet();
+    lineSetSecondary->ref();

```

Comment:
Causes a memory leak because the reference counter will never be decremented to 0.